pkg/csi/service/node.go

```diff
@@ -679,99 +684,51 @@ func (driver *vsphereCSIDriver) NodeGetInfo(
-				NodeId:            nodeID,
-				MaxVolumesPerNode: maxVolumesPerNode,
-			}
-			log.Infof("NodeGetInfo response: %v", nodeInfoResponse)
-			return nodeInfoResponse, nil
-		}
-		return nil, logger.LogNewErrorCodef(log, codes.Internal,
-			"failed to read cnsconfig. err: %v", err)
-	}
-	var accessibleTopology map[string]string
-	topology := &csi.Topology{}
 
-	if cfg.Labels.Zone != "" && cfg.Labels.Region != "" {
-		log.Infof("Config file provided to node daemonset with zones and regions. Assuming topology aware cluster.")
-		vcenterconfig, err := cnsvsphere.GetVirtualCenterConfig(ctx, cfg)
-		if err != nil {
-			return nil, logger.LogNewErrorCodef(log, codes.Internal,
-				"failed to get VirtualCenterConfig from cns config. err: %v", err)
-		}
-		vcManager := cnsvsphere.GetVirtualCenterManager(ctx)
-		vcenter, err := vcManager.RegisterVirtualCenter(ctx, vcenterconfig)
-		if err != nil {
-			return nil, logger.LogNewErrorCodef(log, codes.Internal,
-				"failed to register vcenter with virtualCenterManager. err: %v", err)
+	var (
+		accessibleTopology map[string]string
+		err                error
+	)
+	if commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.ImprovedVolumeTopology) {
+		// Initialize volume topology service.
+		if err = initVolumeTopologyService(ctx); err != nil {
+			return nil, err
 		}
-		defer func() {
-			if vcManager != nil {
-				err = vcManager.UnregisterAllVirtualCenters(ctx)
-				if err != nil {
-					log.Errorf("UnregisterAllVirtualCenters failed. err: %v", err)
-				}
-			}
-		}()
-		//Connect to vCenter
-		err = vcenter.Connect(ctx)
-		if err != nil {
-			return nil, logger.LogNewErrorCodef(log, codes.Internal,
-				"failed to connect to vcenter host: %s. err: %v", vcenter.Config.Host, err)
+		// Fetch topology labels for given node.
+		nodeInfo := csinodetopology.NodeInfo{
+			NodeName: nodeName,
+			NodeID:   nodeID,
 		}
-		// Get VM UUID
-		uuid, err := getSystemUUID(ctx)
-		if err != nil {
-			return nil, logger.LogNewErrorCodef(log, codes.Internal,
-				"failed to get system uuid for node VM. err: %v", err)
-		}
-		log.Debugf("Successfully retrieved uuid:%s  from the node: %s", uuid, nodeID)
-		nodeVM, err := cnsvsphere.GetVirtualMachineByUUID(ctx, uuid, false)
-		if err != nil || nodeVM == nil {
-			log.Errorf("failed to get nodeVM for uuid: %s. err: %+v", uuid, err)
-			uuid, err = convertUUID(uuid)
-			if err != nil {
-				return nil, logger.LogNewErrorCodef(log, codes.Internal,
-					"convertUUID failed with error: %v", err)
-			}
-			nodeVM, err = cnsvsphere.GetVirtualMachineByUUID(ctx, uuid, false)
-			if err != nil || nodeVM == nil {
-				return nil, logger.LogNewErrorCodef(log, codes.Internal,
-					"failed to get nodeVM for uuid: %s. err: %+v", uuid, err)
-			}
+		accessibleTopology, err = topologyService.GetNodeTopologyLabels(ctx, &nodeInfo)
+	} else {
+		// If ImprovedVolumeTopology is not enabled, use the VC credentials to fetch node topology information.
```

Comment:
Can you confirm that there is no change to the existing behavior in this control path?